File: sort/selection_sort.cpp
// +build ignore
// from golang

#include "selection_sort.h"

using namespace std;

// selection sort - desc
void selectionSort(vector<int> &inArr) {
    int size = inArr.size();
    for(int i=0; i<size-1; ++i) {
        for(int z=i; z<size-1; ++z) {
            if (inArr[i] < inArr[z]) {
                int tmp = inArr[i];
                inArr[i] = inArr[z];
                inArr[z] = tmp;
            }
        }
    }
}

File: sort/selection_sort.go
package sort

// selection sort - desc
func SelectionSort(inArr []int) {
	for i := 0; i < len(inArr)-1; i++ {
		for z := i; z < len(inArr)-1; z++ {
			if inArr[i] < inArr[z] {
				inArr[i], inArr[z] = inArr[z], inArr[i]
			}
		}
	}
}
